vcd/datasource_vcd_resource_list.go

```diff
@@ -590,7 +590,18 @@ func orgNetworkListV2(d *schema.ResourceData, meta interface{}) (list []string,
 func getEdgeGatewayList(d *schema.ResourceData, meta interface{}, resType string) (list []string, err error) {
 	client := meta.(*VCDClient)
 
-	org, vdc, err := client.GetOrgAndVdc(d.Get("org").(string), d.Get("vdc").(string))
+	vdcName := d.Get("parent").(string)
+	if vdcName == "" {
+		vdcName = d.Get("vdc").(string)
+	}
+	if vdcName == "" {
+		vdcName = client.Vdc
+	}
+
+	if vdcName == "" {
+		return nil, fmt.Errorf("VDC name not given either as 'vdc' or 'parent' field")
```

Comment:
Maybe you're using these "three ifs to choose at least one of available VDC name fields" in other places too, and it's worth having it in a function? (In an effort to make this code part easier on the eyes).

Reply:
Function added. Some places use it now

Reply:
Neat! Thanks!